Arrowgene.Ddon.GameServer/Handler/CraftStartAttachElementHandler.cs

```diff
@@ -0,0 +1,89 @@
+using Arrowgene.Ddon.GameServer.Characters;
+using Arrowgene.Ddon.Server;
+using Arrowgene.Ddon.Shared.Entity.PacketStructure;
+using Arrowgene.Ddon.Shared.Entity.Structure;
+using Arrowgene.Ddon.Shared.Model;
+using Arrowgene.Logging;
+using Microsoft.VisualBasic;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Runtime.CompilerServices;
+
+namespace Arrowgene.Ddon.GameServer.Handler
+{
+    public class CraftStartAttachElementHandler : GameRequestPacketHandler<C2SCraftStartAttachElementReq, S2CCraftStartAttachElementRes>
+    {
+        private static readonly ServerLogger Logger = LogProvider.Logger<ServerLogger>(typeof(CraftStartAttachElementHandler));
+
+        public CraftStartAttachElementHandler(DdonGameServer server) : base(server)
+        {
+        }
+
+        public override S2CCraftStartAttachElementRes Handle(GameClient client, C2SCraftStartAttachElementReq request)
+        {
+            S2CItemUpdateCharacterItemNtc updateCharacterItemNtc = new S2CItemUpdateCharacterItemNtc();
+
+            var (storageType, itemProps) = client.Character.Storage.FindItemByUIdInStorage(ItemManager.EquipmentStorages, request.EquipItemUId);
+            var (slotNo, item, amount) = itemProps;
+
+
+            ClientItemInfo clientItemInfo = ClientItemInfo.GetInfoForItemId(Server.AssetRepository.ClientItemInfos, item.ItemId);
+            var result = new S2CCraftStartAttachElementRes();
+
+            Character character = null;
+            if (storageType == StorageType.CharacterEquipment)
```

Comment:
Pawns can wear equipment which has crests but I guess currently it can't be modified while the pawn has it equipped.

Reply:
It seems to be possible it just doesnt show the right icon and shows the item bag icon in its place

Reply:
So is this sufficient for now or? None of the queries are returning the character id of the item today. Would need to rewrite a bunch of stuff for this.

Reply:
I'll admit i haven't properly checked. But i'd be surprised if that's the case as that'd put a hard limit of 3 pawns at most while the rest of the pawn system seems to suggest it could easily have no limit.